src/main/repository/PromotionRepository.ts

```diff
@@ -216,6 +225,41 @@ export class PromotionRepository extends Repository<Promotion> {
       return promotion;
     });
   }
+
+  /**
+   * Like as we did in findPromotionUserSaved, check all promotions is voted by the user
+   * @param userId the id of the user
+   * @param promotions the promotions we want to find out if the user voted
+   */
+  private async findPromotionsUserVoted(
+    userId: string,
+    promotions: Promotion[]
+  ): Promise<Promotion[]> {
+    const promotionIds = promotions.map((promotion: Promotion) => promotion.id);
+    const voteRecords = await getConnection()
+      .getCustomRepository(VoteRecordRepository)
+      .find({
+        select: ['promotionId', 'voteState'],
+        where: {
+          userId,
+          promotionId: In(promotionIds),
+        },
+      });
+    // todo: I could not find way to initialize Map using map function, so using for loop
+    const dict = new Map<string, number>(
+      voteRecords.map((voteRecord) => {
+        return [voteRecord['promotionId'], voteRecord['voteState']];
+      })
+    );
+    return promotions.map((promotion: Promotion) => {
+      if (dict.has(promotion.id)) {
+        promotion.voteState = dict.get(promotion.id);
+      } else {
+        promotion.voteState = VoteState.INIT;
+      }
+      return promotion;
+    });
```

Comment:
I think we can simplify the code, how do you feel about this?

```suggestion
    const promotionIdToVoteState = new Map<string, VoteState>(
      voteRecords.map((voteRecord) => {
        return [voteRecord.promotionId, voteRecord.voteState];
      })
    );
    return promotions.map((promotion: Promotion) => {
      promotion.voteState = promotionIdToVoteState.get(promotion.id) ?? VoteState.INIT
      return promotion;
    });
```

Reply:
Oh, thanks for great simplified code!
I'll do that!